packages/opentelemetry-core/test/platform/browser/BasePlugin.test.ts

```diff
@@ -17,24 +17,24 @@
 import {
   NOOP_TRACER,
   NoopTracerProvider,
-  NoopLogger,
+  createNoopDiagLogger,
```

Comment:
Is this exported for any reason other than testing? If not, I would prefer it not be exported from the API package as it then becomes part of the API we have to support.

Reply:
It is mostly for testing (as that is the most usage), to remove would require that all test usages would need to be refactored to not use it (which in most cases they shouldn't need at all). Do we want to do that as part of this PR?

It does raise the following questions
- Do we or should we have some sort of test "helpers" as for this a MockLogger that records all usages in memory would be useful?
- For all existing usages of ```new NoopLogger()``` what do we tell them for upgrading, as there *may* be at least one user that really want to sink all messages?

On the above, where should I add documentation on usage / upgrade notes for the release notes, so that we get people upgrading and using this in the prescribed way?

Reply:
Searching across all usages the only non-api usage is in opentelemetry-shim-opentracing\src\shim.ts which is currently
```this._diagLogger = diagLogger || api.createNoopDiagLogger();```

We could change this to following,m but this would also be a function change -- does anyone know if this would be ok? As it would affect anyone using OpenTracing Tracers (as per the Readme)
```this._diagLogger = diagLogger || diag;```

Reply:
hmm i'm not sure how that would affect the shim as i'm not that familiar with opentracing.

Reply:
Looks like either should be safe as this logger is only used in the shim and not passed to opentracing itself. 

>Do we or should we have some sort of test "helpers" as for this a MockLogger that records all usages in memory would be useful?

I think something like that can be added later. For now I don't think it's something we need to focus on

>For all existing usages of new NoopLogger() what do we tell them for upgrading, as there may be at least one user that really want to sink all messages?

if an end user is passing a NoopLogger into a component, they can just pass a disabled DiagLogger into it. Unless i've missed something?

>On the above, where should I add documentation on usage / upgrade notes for the release notes, so that we get people upgrading and using this in the prescribed way?

in the README.md there is an upgrade guide. You can go ahead and add a new entry.

Reply:
> Looks like either should be safe as this logger is only used in the shim and not passed to opentracing itself.

I've left this alone for now.

> if an end user is passing a NoopLogger into a component, they can just pass a disabled DiagLogger into it. Unless i've missed something?

Correct, you've not missed anything

> in the README.md there is an upgrade guide. You can go ahead and add a new entry.

I've added a new 0.17 -> section to the readme.